### PR TITLE
H-W_1

### PR DESCRIPTION
**Pull request by SalnikovMarkiian**
created Book entity, repo, services.

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/OnlineBookStoreApplication.java [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class OnlineBookStoreApplication {
+    @Autowired
+    private BookService bookService;
```
Comment (VolodymyrSkvira) [inline src/main/java/com/example/onlinebookstore/OnlineBookStoreApplication.java]: You should use a constructor injection. Simply add the @RequiredArgsConstructor annotation above the class and make the field final

File: src/main/java/com/example/onlinebookstore/OnlineBookStoreApplication.java [hunk 2]
```diff
@@ -1,13 +1,30 @@
...
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return new CommandLineRunner() {
+            @Override
+            public void run(String... args) throws Exception {
+
+                System.out.println(bookService.getAll());
+            }
+        };
+    }
```
Comment (VolodymyrSkvira) [inline src/main/java/com/example/onlinebookstore/OnlineBookStoreApplication.java]: Your CommanLineRunner should set all Book entity fields, and then test both methods from your service

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+@Data
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    @Column(unique = true)
+    private String isbn;
+    private BigDecimal price;
```
Comment (VolodymyrSkvira) [inline src/main/java/com/example/onlinebookstore/model/Book.java]:
```suggestion
    private Long id;
    
     @Column(nullable = false)
    private String title;
    
     @Column(nullable = false)
    private String author;
    
    @Column(nullable = false, unique = true)
    private String isbn;
    
     @Column(nullable = false)
    private BigDecimal price;
```

File: src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            throw new DataProcessingException(
+                    "Can`t save book"
+                    + book.getTitle()
+                    + " to DB");
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return book;
```
Comment (VolodymyrSkvira) [inline src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java]: It's time to learn such magic as `inTransaction()` method :)
```suggestion
        try {
            sessionFactory.inTransaction(session -> session.persist(book));
            return book;
        } catch (Exception e) {
            throw new YourExceptionHere("Cannot save new book", e);
        }
```

File: src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> getAllBooksQuery = session.createQuery(
+                    "from Book", Book.class);
+            return getAllBooksQuery.getResultList();
+        }
```
Comment (VolodymyrSkvira) [inline src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java]:
```suggestion
      try (Session session = sessionFactory.openSession()) {
            return session.createQuery("from Book", Book.class).getResultList();
        } catch (Exception e) {
            throw new YourExceptionHere("Cannot get all books", e);
        }
```

File: src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
```
Comment (VolodymyrSkvira) [inline src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java]: You should use a constructor injection. Simply add the @requiredargsconstructor annotation above the class and make the field final

**Review by VolodymyrSkvira — CHANGES_REQUESTED** [review]
Great solution, time to change some things :)

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    private String description;
+    private String coverImage;
+
```
Comment (Alexsandr-Yablunovskyi) [inline src/main/java/com/example/onlinebookstore/model/Book.java]: Empty line

**Review by Alexsandr-Yablunovskyi — APPROVED** [review]
Great job!

File: src/main/java/com/example/onlinebookstore/model/Book.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
+@Data
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/model/Book.java:13]: Add Table annotation with name

File: src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@RequiredArgsConstructor
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/onlinebookstore/repository/BookRepositoryImpl.java:14]: To be consistent use constructor as you do in BookServiceImpl

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

**Pull request merged**